sdk/storage/Azure.Storage.Common/src/OperationInterruptedException.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.Serialization;
+using System.Text;
+using Azure.Core;
+
+namespace Azure.Storage
+{
+#pragma warning disable CA2229, CA2235 // False positive
+    /// <summary>
+    /// An exception thrown when an operation is interrupted and can be continued later on.
+    /// </summary>
+    [Serializable]
+    public class OperationInterruptedException : RequestFailedException, ISerializable
+    {
+        /// <summary>
+        /// Gets the continuation token from the last successful response from the service.
+        /// </summary>
+        public string ContinuationToken { get; }
```

Comment:
We shouldn't create an exception this general for a purpose this specific.  Both the name and the assembly should be made DataLake specific.  I don't think we should extend `RequestFailedException` because this could be triggered by anything.

Instead of a new type, I'd rather just throw an `InvalidOperationException` that
- has a RequestFailedException as its InnerException,
- sets `ex.Data[$"{OperationName}.ContinuationToken"] = token`,
- and has a message that talks about `$"resuming with continuation token xsdfajsdjasdfkjaskdjfasdf, which can also be obtained from Exception.Data["{OperationName}.ContinuationToken"].`

This might be worth another quick meeting with Krzysztof to make sure we're getting it right.  I don't think we've done anything else like this in Track 2 yet.

Reply:
How about `DatalakeChangeAccessControlRecursiveInterruptedException` or so that has innerexception and token?
Since we want to encourage customers to do proper error handling for this API (since client side is responsible for end result if we have multiple batches) then I'd rather not use `InvalidOperationException`  - one might be under impression that it could be thrown under other circumstances also, thus pushing people to write more exception filtering logic (i.e. checking for Exception.Data content.
Documentation wise it's also easier to say "catch DatalakeChangeAccessControlRecursiveInterruptedException and see what's under ContinuationToken" instead of "catch InvalidOperationException then inspect that specific key under Exception.Data". 
Extensibility wise - adding properties to custom exception type and to Exception.Data is similar, but properties are compiler enforced contract.

Reply:
I think this scenario justifies a custom exception. I would just make the name a bit simpler. We don't want to look like [this APIs](https://docs.microsoft.com/en-us/dotnet/api/system.servicemodel.messagesecurityversion.wssecurity11wstrustfebruary2005wssecureconversationfebruary2005wssecuritypolicy11basicsecurityprofile10?view=dotnet-plat-ext-3.1). Maybe DataLakeAclChangeFailedException?